addons/website_event/tests/test_event_internals.py

```diff
@@ -224,3 +226,87 @@ def test_registration_answer_search(self):
+        self.assertEqual([True, False, True], events_can_see_compute)
+
+        events_user_can_see_search = events_rset._search_is_visible_on_website('=', True)[0][2]
+        self.assertTrue(event_public.id in events_user_can_see_search)
+        self.assertTrue(event_portal.id in events_user_can_see_search)
+        self.assertTrue(event_link.id not in events_user_can_see_search)
+
+    @users('admin')
+    def test_admin_user_is_visible_on_website(self):
+        """
+        Tests the computed field `is_visible_on_website` on `event.event`.
+        `filtered_domain` is used to check if the `_compute_is_visible_on_website` method is correct
+        `search` is used to check if the `_search_is_visible_on_website` method is correct
+        """
+
+        event_public, event_link, event_portal = [
+            self.env['event.event'].sudo().create(
+                {'name': 'event', 'website_visibility': website_visibility}
+            ) for website_visibility in ['public', 'link', 'logged_users']
+        ]
+        events_rset = event_public | event_link | event_portal
+
+        events_rset._compute_is_visible_on_website()
+        events_can_see_compute = events_rset.mapped('is_visible_on_website')
+        self.assertEqual([True, False, True], events_can_see_compute)
+
+        events_user_can_see_search = events_rset._search_is_visible_on_website('=', True)[0][2]
+        self.assertTrue(event_public.id in events_user_can_see_search)
+        self.assertTrue(event_portal.id in events_user_can_see_search)
+        self.assertTrue(event_link.id not in events_user_can_see_search)
+
+    @users('public')
+    def test_public_user_is_visible_on_website(self):
+        """
+        Tests the computed field `is_visible_on_website` on `event.event`.
+        `filtered_domain` is used to check if the `_compute_is_visible_on_website` method is correct
+        `search` is used to check if the `_search_is_visible_on_website` method is correct
+        """
+
+        event_public, event_link, event_portal = [
+            self.env['event.event'].sudo().create(
+                {'name': 'event', 'website_visibility': website_visibility}
+            ) for website_visibility in ['public', 'link', 'logged_users']
+        ]
+        events_rset = event_public | event_link | event_portal
+
+        events_rset._compute_is_visible_on_website()
+        events_can_see_compute = events_rset.mapped('is_visible_on_website')
+        self.assertEqual([True, False, False], events_can_see_compute)
+
+        events_user_can_see_search = events_rset._search_is_visible_on_website('=', True)[0][2]
+        self.assertTrue(event_public.id in events_user_can_see_search)
+        self.assertFalse(event_portal.id in events_user_can_see_search)
+        self.assertFalse(event_link.id in events_user_can_see_search)
+
+        # users can see events if they are participating
+        with patch.object(Event, '_fetch_is_participating_events', lambda s: events_rset):
```

Comment:
Would be closer to a real use case to create a visitor record matching our user and making it participate to the event.